editor/animation_track_editor.cpp

```diff
@@ -5300,11 +5331,95 @@ void AnimationTrackEditor::_new_track_node_selected(NodePath p_path) {
+	if (pick_move_track->is_connected("selected", callable_mp(this, &AnimationTrackEditor::_move_track_to_node))) {
+		pick_move_track->disconnect("selected", callable_mp(this, &AnimationTrackEditor::_move_track_to_node));
+	}
+	pick_move_track->connect("selected", callable_mp(this, &AnimationTrackEditor::_move_track_to_node).bind(p_track));
+}
+
+/*
+	Takes p_path - a NodePath to a new target node (NOT with the property)
+	and p_track - The AnimationTrack
+	Changes the Animation's path to match that node.
+*/
+void AnimationTrackEditor::_move_track_to_node(NodePath p_path, int p_track) {
+	// Get new_node
+	Node *tree_root = EditorNode::get_singleton()->get_tree()->get_root();
+	Node *new_node = tree_root->get_node_or_null(p_path);
+	ERR_FAIL_NULL(new_node);
+
+	// Get AnimationPlayer's root_node
+	AnimationPlayer *ap = AnimationPlayerEditor::get_singleton()->get_player();
+	if (!ap) {
+		ERR_FAIL_EDMSG("No AnimationPlayer is currently being edited.");
+	}
+	Node *root_node = ap->get_node_or_null(ap->get_root_node());
+	if (!root_node) {
+		EditorNode::get_singleton()->show_warning(TTR("Not possible to change node path without a root"));
+		return;
+	}
+
+	// Get New Path, in respect to AP's root_node
+	String current_track_path = animation->track_get_path(p_track);
+	String property_path = current_track_path.get_slice(":", 1);
+	String new_path = root_node->get_path_to(new_node);
+	new_path = new_path + ":" + property_path;
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Change Track Path"));
+	undo_redo->add_do_method(animation.ptr(), "track_set_path", p_track, new_path);
+	undo_redo->add_undo_method(animation.ptr(), "track_set_path", p_track, current_track_path);
+	undo_redo->commit_action();
+}
+
+void AnimationTrackEditor::_change_track_property_selected(const String &p_name, int p_track) {
+	String current_track_path = animation->track_get_path(p_track);
+	String base_path = current_track_path.get_slice(":", 0);
+	String new_path = base_path + ":" + p_name;
+
+	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+	undo_redo->create_action(TTR("Change Track Path"));
+	undo_redo->add_do_method(animation.ptr(), "track_set_path", p_track, new_path);
+	undo_redo->add_undo_method(animation.ptr(), "track_set_path", p_track, current_track_path);
+	undo_redo->commit_action();
+}
+
+void AnimationTrackEditor::_change_track_target_property_pressed(int p_track) {
+	AnimationPlayer *ap = AnimationPlayerEditor::get_singleton()->get_player();
+	if (!ap) {
+		ERR_FAIL_EDMSG("No AnimationPlayer is currently being edited.");
+	}
+	NodePath current_path = animation->track_get_path(p_track);
+	Node *root_node = ap->get_node_or_null(ap->get_root_node());
+	Node *current_node = root_node->get_node_or_null(animation->track_get_path(p_track));
+	if (!current_node) {
+		EditorNode::get_singleton()->show_warning(TTR("Not possible to change property without a valid target node"));
```

Comment:
If it's not too complicated, you could instead disable the menu item if the node is invalid.

Reply:
I've disabled the menu when the node is invalid (and also when the track type does not support properties). This error message might be redundant as a result.

>Also technically referring to "node" is incorrect, as the target can be any Object.

I could be missing an edge case, but I imagine this refers to when pointing to an object referenced by a node like `Node3D/Sprite3D:material_override:shader` references a material_override object, and the shader within that object.

This particular error message checks for the node in the path (In this case Sprite3D the child of Node3D) without regard to the properties. It _sounds_ correct to me but let me know if I'm missing something.